frontend/src/container/GridTableComponent/index.tsx

```diff
@@ -52,6 +53,35 @@ function GridTableComponent({
 		[columns],
 	);
 
+	const applyColumnUnits = useCallback(
+		(dataSource: RowData[]): RowData[] => {
+			let mutateDataSource = dataSource;
+			if (isEmpty(columnUnits)) {
+				return mutateDataSource;
+			}
+
+			mutateDataSource = mutateDataSource.map(
+				(val): RowData => {
+					const newValue = val;
+					Object.keys(val).forEach((k) => {
+						if (columnUnits[k]) {
+							newValue[k] = getYAxisFormattedValue(String(val[k]), columnUnits[k]);
+						}
+					});
+					return newValue;
+				},
+			);
```

Comment:
I believe this mutation is buggy https://www.loom.com/share/f761278edefc40c0ba6577210e0fc12c

Reply:
you do get the correct results on stage and run query. let me check why this is not updating...

Reply:
fixed the same. @srikanthccv 


https://github.com/SigNoz/signoz/assets/54737045/70f84ef2-23f9-481a-8e54-ce6c1a36a08d